src/pages/add/import-wallet/import-wallet.ts

```diff
@@ -85,19 +120,259 @@ export class ImportWalletPage implements OnInit {
+        this.onGoingProcessProvider.set('importingWallet', false);
+        this.finish(wallet);
+      }).catch((err: any) => {
+        if (err instanceof this.errors.NOT_AUTHORIZED) {
+          this.importErr = true;
+        } else {
+          this.popupProvider.ionicAlert('Error', err); // TODO: gettextcatalog
+        }
+      });
+    }, 100);
+  }
+
 
   import() {
-    console.log(this.formData);
+    if (this.selectedTab === 'file') {
+      this.importFromFile();
+    } else {
+      this.importFromMnemonic();
+    }
   }
+
+  public importFromFile(): void {
+    if (!this.importForm.valid) {
+      this.popupProvider.ionicAlert('Error', 'There is an error in the form'); // TODO: gettextcatalog
+      return;
+    }
+
+    let backupFile = this.file;
+    let backupText = this.formData.backupText;
+    let password = this.formData.filePassword;
+
+    if (!backupFile && !backupText) {
+      this.popupProvider.ionicAlert('Error', 'Please, select your backup file'); // TODO: gettextcatalog
+      return;
+    }
+
+    if (backupFile) {
+      this.reader.readAsBinaryString(backupFile);
+    } else {
+      let opts: any = {};
+      opts.bwsurl = this.formData.bwsurl;
+      opts.coin = this.formData.coin;
+      this.importBlob(backupText, opts);
+    }
+  }
+
+  public importFromMnemonic(): void {
+    if (!this.importForm.valid) {
+      this.popupProvider.ionicAlert('Error', 'There is an error in the form'); // TODO: gettextcatalog
+      return;
+    }
+
+    let opts: any = {};
+
+    if (this.formData.bwsurl)
+      opts.bwsurl = this.formData.bwsurl;
+
+    let pathData: any = this.derivationPathHelperProvider.parse(this.formData.derivationPath);
+
+    if (!pathData) {
+      this.popupProvider.ionicAlert('Error', 'Invalid derivation path'); // TODO: gettextcatalog
+      return;
+    }
+
+    opts.account = pathData.account;
+    opts.networkName = pathData.networkName;
+    opts.derivationStrategy = pathData.derivationStrategy;
+    opts.coin = this.formData.coin;
+
+    let words: string = this.formData.words || null;
+
+    if (!words) {
+      this.popupProvider.ionicAlert('Error', 'Please enter the recovery phrase');
+      return;
+    } else if (words.indexOf('xprv') == 0 || words.indexOf('tprv') == 0) {
+      return this.importExtendedPrivateKey(words, opts);
+    } else if (words.indexOf('xpub') == 0 || words.indexOf('tpuv') == 0) {
+      //return this.importExtendedPublicKey(words, opts); TODO
+      return this.logger.warn("TODO: this.importExtendedPublicKey(words, opts)");
+    } else {
+      let wordList: Array<any> = words.split(/[\u3000\s]+/);
+
+      if ((wordList.length % 3) != 0) {
+        this.popupProvider.ionicAlert('Error', 'Wrong number of recovery words: ' + wordList.length);
+        return;
+      }
+    }
+
+    opts.passphrase = this.formData.passphrase || null;
+    this.importMnemonic(words, opts);
+  }
+
+  public toggleShowAdvOpts(): void {
```

Comment:
If this function only switch to true/false should be here?

Reply:
I prefer to have this type of functions in the controller, to not dirty the html